include/toolchain/gcc.h

```diff
@@ -194,6 +194,10 @@ do {                                                                    \
 
 #define popcount(x) __builtin_popcount(x)
 
+#ifndef __no_optimization
+#define __no_optimization __attribute__((optimize("-O0")))
+#endif
```

Comment:
Good to have this extra abstraction instead of assuming '-O0' disables optimization for all platforms.
Did you cover all toolchains (don't see a define for xcc)? Should there be a default (-O0) in toolchain/common.h and only overrides for compilers if needed?

Reply:
> Did you cover all toolchains (don't see a define for xcc)?

Yes. I've even added the note about that to commit message:

> NOTE: we don't define __no_optimization for XCC as it includes GCC header with __no_optimization defined.

IOW: all toolchains are covered.

Reply:
Thanks, missed the individual commit remarks.